woocommerce/woocommerce-ios
# [Backend Receipts] Analytics

Closes #11856
Its better if we review/merge https://github.com/woocommerce/woocommerce-ios/issues/11829 first, but not a blocker.

## Description
This PR updates analytics for receipt-related events. Now they'll track an additional property `source: local | backend` depending if is generated locally or from the backend.

## Changes:
* Adds "source" property to receipt events. This property will mark the event source as `local` if we're dealing with a legacy receipt, or `backend` if is a backend-generated receipt.
* Updates some of the existing properties to be optional, since we may have no `CountryCode` or when a receipt is generated via a non-card present method.

## Testing instructions
There's no need to test all cases, since some flows might require a bit of setup (eg: The print button is only enabled if you use the printer simulator from Xcode additional tools, but sharing is enabled by default), just go through the receipts flow, attempt to print/share, and observe that `source: local | backend` is logged. 

Examples of some flows that can be checked:
- With the feature flag on, print or share cash receipt.
```
Tracked receipt_print_success, properties: [AnyHashable("country"): "US", AnyHashable("is_wpcom_store"): false, AnyHashable("was_ecommerce_trial"): false, AnyHashable("source"): "local", AnyHashable("plan"): "jetpack_free", AnyHashable("blog_id"): 229034549, AnyHashable("store_id"): "59682e4b-5967-4a66-b8dc-831526e293b3", AnyHashable("site_url"): "https://loudly-silly-kid.jurassic.ninja"]
```
- With the feature flag on, share card present receipt.
```
Tracked receipt_email_success, properties: [AnyHashable("site_url"): "https://loudly-silly-kid.jurassic.ninja", AnyHashable("source"): "backend", AnyHashable("blog_id"): 229034549, AnyHashable("store_id"): "59682e4b-5967-4a66-b8dc-831526e293b3", AnyHashable("was_ecommerce_trial"): false, AnyHashable("plan"): "jetpack_free", AnyHashable("is_wpcom_store"): false]
```
- Cancelling printing tracks the event.
```
Tracked receipt_print_canceled, properties: [AnyHashable("was_ecommerce_trial"): false, AnyHashable("source"): "local", AnyHashable("plan"): "jetpack_free", AnyHashable("is_wpcom_store"): false, AnyHashable("blog_id"): 229034549, AnyHashable("country"): "US", AnyHashable("store_id"): "59682e4b-5967-4a66-b8dc-831526e293b3", AnyHashable("site_url"): "https://loudly-silly-kid.jurassic.ninja"]
```
- Forcing an error logs the error:
```
Tracked receipt_print_failed, properties: [AnyHashable("plan"): "jetpack_free", AnyHashable("is_wpcom_store"): false, AnyHashable("store_id"): "59682e4b-5967-4a66-b8dc-831526e293b3", AnyHashable("error_description"): "Error Domain= Code=0 \"(null)\"", AnyHashable("error_domain"): "", AnyHashable("was_ecommerce_trial"): false, AnyHashable("site_url"): "https://loudly-silly-kid.jurassic.ninja", AnyHashable("error_code"): "0", AnyHashable("source"): "backend", AnyHashable("blog_id"): 229034549]
```
